test/_files/expected-factories/with-params.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * Generated factory for ZendTest\Di\TestAsset\Constructor\MixedArguments
+ */
+
+namespace ZendTest\Di\Generated\Factory\ZendTest\Di\TestAsset\Constructor;
+
+use Psr\Container\ContainerInterface;
+use Zend\Di\CodeGenerator\FactoryInterface;
+use function array_key_exists;
+use function is_array;
+
+final class MixedArgumentsFactory implements FactoryInterface
+{
```

Comment:
This is how the output should finally look like. To achieve this I'd remove zend-code. This simplifies the generator, docs and consumers do no longer need zend-code for AoT